### ISSUE TITLE
Tracking the API bug (mismatch between /TR and API for "vocab-dcat")

### ISSUE THREAD

**Issue by riccardoAlbertoni** [w3c/dxwg#1223]
This issue tracks the API bug [w3c/w3c-api#96](https://github.com/w3c/w3c-api/issues/96), which impacts on the mismatch between /TR and API for "vocab-dcat". 

This is the part still pending from the closure of issue #1177.

**Comment by riccardoAlbertoni** [comment]
The API bug has been addressed. Any objections to close this issue?

**Comment by andrea-perego** [comment]
I guess so. However, AFAIK, the inconsistency in SpecRef has not been addressed - see https://github.com/w3c/dxwg/issues/1177#issuecomment-577680178. Namely

https://api.specref.org/bibrefs?refs=vocab-dcat

still returns the entry for DCAT 2014.

**Comment by plehegar** [comment]
The issue is indeed ongoing. As of 2 weeks ago, the W3C API started to include versioning information. For example, see the "[Versioning section](https://www.w3.org/PM/Groups/chairboard.html?gid=99375#series)" in the chairboard. The information is not yet authoritative however, meaning that https://www.w3.org/TR doesn't use it yet.

**Comment by andrea-perego** [comment]
@plehegar , I wonder whether there's any update on this issue.

The SpecRef API is returning inconsistent entries when using the `vocab-dcat` keyword - title and editors are the ones of the 2014 version of DCAT, but the date is 4 Feb 2020.

See: https://api.specref.org/bibrefs?refs=vocab-dcat

````json
{
   "vocab-dcat":{
      "authors":[
         "Fadi Maali",
         "John Erickson"
      ],
      "href":"https://www.w3.org/TR/vocab-dcat/",
      "title":"Data Catalog Vocabulary (DCAT)",
      "status":"REC",
      "publisher":"W3C",
      "deliveredBy":[
         {
            "url":"https://www.w3.org/2011/gld/",
            "shortname":"gld"
         }
      ],
      "versions":[
         "vocab-dcat-20200204",
         "vocab-dcat-20140116",
         "vocab-dcat-20131217",
         "vocab-dcat-20131105",
         "vocab-dcat-20130801",
         "vocab-dcat-20130312",
         "vocab-dcat-20120405"
      ],
      "obsoletedBy":[
         "vocab-dcat-2"
      ],
      "id":"vocab-dcat",
      "date":"4 February 2020"
   }
}
````

**Comment by andrea-perego** [comment]
Just noting that this issue is yet to be fixed - see https://github.com/tobie/specref/issues/640

**Comment by andrea-perego** [comment]
@plehegar , could you please provide an update about this issue?

**Comment by andrea-perego** [comment]
> @plehegar , I wonder whether there's any update on this issue.
> 
> The SpecRef API is returning inconsistent entries when using the `vocab-dcat` keyword - title and editors are the ones of the 2014 version of DCAT, but the date is 4 Feb 2020.
> 
> See: https://api.specref.org/bibrefs?refs=vocab-dcat

Just noticed there's the same problem with JSON-LD. The following URL:

https://www.w3.org/TR/json-ld/

returns the JSON-LD 1.1 spec. However, when using the `lson-ld` keyword, the SpecRef API returns the entry for JSON-LD 1.0, but with the date of the JSON-LD 1.1 spec (16 July 2020).

See https://api.specref.org/bibrefs?refs=json-ld

BTW, @riccardoAlbertoni , should we explicitly refer to JSON-LD 1.1 in DCAT3?

**Comment by pchampin** [comment]
@andrea-perego the reason why JSON-LD 1.0 has the same date as JSON-LD 1.1 is that JSON-LD 1.0 was marked superceded last modified when JSON-DL 1.1 was published. This is **not** a bug.

The same applies to vocab-dcat.

**Comment by andrea-perego** [comment]
@pchampin said:

> @andrea-perego the reason why JSON-LD 1.0 has the same date as JSON-LD 1.1 is that JSON-LD 1.0 was marked superceded last modified when JSON-DL 1.1 was published. This is **not** a bug.
> 
> The same applies to vocab-dcat.

The bug I was referring to is not about the date (which is indeed correct) but about the title and editors, that are still those of version 1 (for both DCAT and JSON-LD), whereas they should be the ones of the latest REC.

**Comment by pchampin** [comment]
@andrea-perego sorry for letting this linger for so long.

Let me try to summarize our findings so fat:
* on w3.org/TR, [`vocab-dcat`](https://www.w3.org/TR/vocab-dcat/) (without any version number) points to the *latest* REC (currently `vocab-dcat-2`),
* wheras on specref, all the fields of  [`vocab-dcat`](https://api.specref.org/bibrefs?refs=vocab-dcat) are describing the *first* version of DCAT, except for `href` which is pointing to the "versionless" URL https://www.w3.org/TR/vocab-dcat/ .

So we have two issues:
* specref is not self-consistent in the description of version-less specs ([`vocab-dcat`](https://api.specref.org/bibrefs?refs=vocab-dcat), but also [`json-ld`](https://api.specref.org/bibrefs?refs=json-ld), and possibly others)
* specref is inconsistent with TR when it comes to version-less names

Do we agree?

**Comment by pchampin** [comment]
Incidentally, this issue seems related to tobie/specref#463, which at the time (2018) was considered too much work to fix...

**Comment by andrea-perego** [comment]
@pchampin said:

> @andrea-perego sorry for letting this linger for so long.
> 
> Let me try to summarize our findings so fat:
> 
> * on w3.org/TR, [`vocab-dcat`](https://www.w3.org/TR/vocab-dcat/) (without any version number) points to the _latest_ REC (currently `vocab-dcat-2`),
> * wheras on specref, all the fields of  [`vocab-dcat`](https://api.specref.org/bibrefs?refs=vocab-dcat) are describing the _first_ version of DCAT, except for `href` which is pointing to the "versionless" URL https://www.w3.org/TR/vocab-dcat/ .
> 
> So we have two issues:
> 
> * specref is not self-consistent in the description of version-less specs ([`vocab-dcat`](https://api.specref.org/bibrefs?refs=vocab-dcat), but also [`json-ld`](https://api.specref.org/bibrefs?refs=json-ld), and possibly others)
> * specref is inconsistent with TR when it comes to version-less names
> 
> Do we agree?

Yes, but with a difference:

> * on w3.org/TR, [`vocab-dcat`](https://www.w3.org/TR/vocab-dcat/) (without any version number) points to the _latest_ REC (currently `vocab-dcat-2`),
> * wheras on specref, all the fields of  [`vocab-dcat`](https://api.specref.org/bibrefs?refs=vocab-dcat) are describing the _first_ version of DCAT, except for `href` which is pointing to the "versionless" URL https://www.w3.org/TR/vocab-dcat/ .

and except for `rawDate`, which corresponds to the one of the latest REC (`2020-02-04`).

This is the record for `vocab-dcat` in SpecRef (https://github.com/tobie/specref/blob/main/refs/w3c.json):

````json
    "vocab-dcat": {
        "authors": [
            "Fadi Maali",
            "John Erickson"
        ],
        "href": "https://www.w3.org/TR/vocab-dcat/",
        "title": "Data Catalog Vocabulary (DCAT)",
        "rawDate": "2020-02-04",
        "status": "REC",
        "publisher": "W3C",
        "deliveredBy": [
            "https://www.w3.org/2011/gld/"
        ],
        "source": "https://www.w3.org/2002/01/tr-automation/tr.rdf",
        "versions": {
          ...
        },
        "isRetired": true,
        "obsoletedBy": [
            "vocab-dcat-2"
        ]
    },
````

which should be modified as follows:

````diff
    "vocab-dcat": {
        "authors": [
-            "Fadi Maali",
-            "John Erickson"
+            "Riccardo Albertoni",
+            "David Browning",
+            "Simon Cox",
+            "Alejandra Gonzalez Beltran",
+            "Andrea Perego",
+            "Peter Winstanley"
        ],
        "href": "https://www.w3.org/TR/vocab-dcat/",
-        "title": "Data Catalog Vocabulary (DCAT)",
+        "title": "Data Catalog Vocabulary (DCAT) - Version 2",
        "rawDate": "2020-02-04",
        "status": "REC",
        "publisher": "W3C",
        "deliveredBy": [
-            "https://www.w3.org/2011/gld/"
+            "https://www.w3.org/2017/dxwg/"
        ],
        "source": "https://www.w3.org/2002/01/tr-automation/tr.rdf",
        "versions": {
          ...
        },
-        "isRetired": true,
-        "obsoletedBy": [
-            "vocab-dcat-2"
-        ]
    },
````

Besides this, also the entries in the `versions` field should be modified for `vocab-dcat`. 

Currently, the record in SpecRef lists as versions all the versions of `vocab-dcat-1` until REC. However, if `vocab-dcat` is meant to return only the latest REC, its versions should be instead `vocab-dcat-1` and `vocab-dcat-2`.

So, it should be something like

````json
        "versions": {
          "20140116" : {
            "aliasOf": "vocab-dcat-1"
          },
          "20200204" : {
            "aliasOf": "vocab-dcat-2"
          }
        }
````

**Comment by pchampin** [comment]
> and except for `rawDate`, which corresponds to the one of the latest REC (`2020-02-04`).

No, this rawDate is also consistent with vocab-dcat-1: this is the last modification date of the REC, when it was retired (because vocab-dcat-2 was published). See https://www.w3.org/standards/history/vocab-dcat-1/

> However, **if** vocab-dcat is meant to return only the latest REC (...)

And it is that "if" that I need to clarify. My understanding from tobie/specref#463 is that it is not a trivial change. At least, it wasn't in 2018 -- maybe things are better now.
